player/lua/stats.lua

```diff
@@ -502,41 +494,40 @@ local function add_video(s)
     append_display_sync(s)
     append_perfdata(s, o.print_perfdata_passes)
 
-    if append_property(s, "video-params/w", {prefix="Native Resolution:"}) then
-        append_property(s, "video-params/h",
-                        {prefix="x", nl="", indent=" ", prefix_sep=" ", no_prefix_markup=true})
+    if append(s, r["w"], {prefix="Native Resolution:"}) then
+        append(s, r["h"], {prefix="x", nl="", indent=" ", prefix_sep=" ", no_prefix_markup=true})
     end
     append_property(s, "window-scale", {prefix="Window Scale:"})
-    append_property(s, "video-params/aspect", {prefix="Aspect Ratio:"})
-    append_property(s, "video-params/pixelformat", {prefix="Pixel Format:"})
+    append(s, format("%.2f", r["aspect"]), {prefix="Aspect Ratio:"})
+    append(s, r["pixelformat"], {prefix="Pixel Format:"})
 
     -- Group these together to save vertical space
-    local prim = append_property(s, "video-params/primaries", {prefix="Primaries:"})
-    local cmat = append_property(s, "video-params/colormatrix",
-                                     {prefix="Colormatrix:", nl=prim and "" or o.nl})
-    append_property(s, "video-params/colorlevels", {prefix="Levels:", nl=cmat and "" or o.nl})
+    local prim = append(s, r["primaries"], {prefix="Primaries:"})
+    local cmat = append(s, r["colormatrix"], {prefix="Colormatrix:", nl=prim and "" or o.nl})
+    append(s, r["colorlevels"], {prefix="Levels:", nl=cmat and "" or o.nl})
 
     -- Append HDR metadata conditionally (only when present and interesting)
-    local hdrpeak = mp.get_property_number("video-params/sig-peak", 0)
+    local hdrpeak = r["sig-peak"] or 0
     local hdrinfo = ""
     if hdrpeak > 1 then
-        hdrinfo = " (HDR peak: " .. hdrpeak .. ")"
+        hdrinfo = " (HDR peak: " .. format("%.2f", hdrpeak) .. ")"
     end
 
-    append_property(s, "video-params/gamma", {prefix="Gamma:", suffix=hdrinfo})
+    append(s, r["gamma"], {prefix="Gamma:", suffix=hdrinfo})
     append_property(s, "packet-video-bitrate", {prefix="Bitrate:", suffix=" kbps"})
     append_filters(s, "vf", "Filters:")
 end
 
 
 local function add_audio(s)
-    if not has_audio() then
+    local r = mp.get_property_native("audio-out-params")
```

Comment:
Are we supposed to be pushing out the information on what is going to the AO or what the input is? The bit rate would let me guess that we want the input info?

`audio-out-params` will have stereo with `--audio-channels=stereo`, while using `audio-params` will get you the input channel count. Of course we could have two separate entries for Input and AO audio parameters.

Reply:
Video seems to be less affected with `vo=gpu´ at least, so in that sense for video `video-out-params` makes sense.

Reply:
`audio-out-params` seems to be more or less the same as `audio-params` according to the manual and my check with non-stereo input showing non-stereo in both. Other than "format", they're the same whether or not you use --audio-channels, at least with WASAPI.

With `--audio-channels=stereo` with non-stereo input, both show 2 channels.

To actually show the proper number of channels would need `track-list` instead, afaik.

Reply:
I'm seeing them as different on alsa. non-stereo input, stereo output even when I specify `ad-lavc-downmix=yes` on a dca codec. 

Not a huge deal though, if anything I think showing the output parameters is more interesting. Although, showing both would be my preference

Reply:
The stance regarding "what stats to show" has been "attributes of the file, not what the player does with it" so far. Reasoning is usually: you know what the player does (as you specified it via arguments), but you don't know the properties of your files. I've always seen this as a kind of mediainfo-lite within the player. However, this is not entirely strict as in some cases both information is useful/makes sense. Adding both, input and output channel count, is certainly possible if desired.

In this case, however, I was not aware of the implication of this change so it was not intentional (although of course it makes sense, thinking about the `out` in the property names...).
@jeeb you say video is less affected. What about stuff like color matrix? Would it, due to this change, now show the user-selected color matrix instead what was specified in the file's metadata? Because it sounds like that to me. And so far we've explicitly shown what the file specified and even declined requests to show the actually used matrix in the past with the reasoning above (-> when you change the matrix then you know you did it, therefore, no need to display it).

Okay so this problem can be fixed by changing `video|audio-out-params` to `video|audio-params`. However, 'detection' of video/audio will still need `video|audio-out-params` to cover the lavfi-complex case. Although in that case `video|audio-params` is probably unavailable and only very little information would be shown. `video|audio-out-params` could be used as fallback when `video|audio-params` is unavailable but then semantics of the shown information implicitly changes without the user noticing. Not exactly good imo.

Reply:
Maybe get `track-list` and use that as default then followed by `video|audio-params`, then lastly `video|audio-out-params` as fallback? The last two have the same keys, so they could be used interchangeably.

Reply:
What good would `track-list` do me? `video|audio-out-params` as fallback for `video|audio-params` would be trivial, it's just the "semantics change" thing here. 
Although after thinking about it again it doesn't matter much. In the case of the issue (that lead to this PR) there is no input file anyway, so output is all we have and unless we want to display nothing, we display out values.
I will do that when I'm near a computer again.

Reply:
> @jeeb you say video is less affected. What about stuff like color matrix? Would it, due to this change, now show the user-selected color matrix instead what was specified in the file's metadata? Because it sounds like that to me.

Yup, I just worded it badly as I was quite tired at that point in the night :D It *generally* doesn't get affected unless there has to be extra filtering done before VO. Any additional filtering added either implicitly by mpv's logic or explicit user commands will of course be visible for both video and audio -out-params. Thus if the idea is to give what's there in the input, you pretty much want to utilize the one without -out.

Thus it does sound good enough to first try getting -params, and then fall back onto -out-params if we have no real "input".